Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/RideSummary.hs

```diff
@@ -0,0 +1,76 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingStrategies #-}
+
+module Domain.Action.UI.RideSummary where
+
+import Data.Time (Day, addDays)
+import Domain.Types.DailyStats as DDS
+import qualified Domain.Types.Merchant as Merchant
+import qualified Domain.Types.Person as DP
+import Kernel.Prelude
+import Kernel.Types.Common
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import qualified Storage.CachedQueries.DailyStats as CQDS
+import qualified Storage.Queries.DailyStats as SQDS
+
+data DriverRideSummaryResp = DriverRideSummaryResp
+  { earnings :: Maybe Money,
+    rideDistance :: Meters,
+    rideDate :: Day,
+    noOfRides :: Int
+  }
+  deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
+
+newtype DriverRideSummaryListResp = DriverRideSummaryListResp
+  { list :: [DriverRideSummaryResp]
+  }
+  deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
+
+listDriverRidesSummary :: (CacheFlow m r, EsqDBFlow m r, MonadFlow m) => Id DP.Person -> Id Merchant.Merchant -> Day -> Day -> m DriverRideSummaryListResp
+listDriverRidesSummary personId _ fromDay toDay = do
+  let allDates = datesBetween fromDay toDay
+  dailyStatsList <- mapM (`findStatsForDate` personId) allDates
+  let list = mkRideSummaryList dailyStatsList
+  return $ DriverRideSummaryListResp {..}
+
+findStatsForDate :: (CacheFlow m r, EsqDBFlow m r) => Day -> Id DP.Person -> m DDS.DailyStats
+findStatsForDate day driverId = do
+  maybeDailyStats <- CQDS.getDriverStatsByDay driverId day
+  case maybeDailyStats of
+    Just a -> pure a
+    Nothing -> fetchDailyStatsfromKV day driverId
+
+fetchDailyStatsfromKV :: (CacheFlow m r, EsqDBFlow m r) => Day -> Id DP.Person -> m DDS.DailyStats
+fetchDailyStatsfromKV day driverId = do
+  mbDailyStats <- SQDS.findByDriverIdAndDate driverId day
+  case mbDailyStats of
+    Just a -> pure a
+    Nothing -> pure $ DDS.DailyStats {id = "", driverId = driverId, totalEarnings = 0, numberOfRides = 0, totalDistance = 0, ofDate = day}
```

Comment:
`Maybe DailyStats` can be sent from here and `catMaybe` can be used by the caller instead of sending dummy data

Reply:
the reason i am sending sort of a dummydata with date here is because if I send data after using catmaybe then I will have to handle it afterwards searching for each date and then use that value and if that date is not present then I have to do the same.

Reply:
Don't know about frontend code but from the above statement it looks like you'll always assume fixed number of elements in response. Will UI break if `toDate - fromDate > 28`

Reply:
we can avoid any restriction of dates `to` `from` range. whatever this query gets should return back, with some basic sanity of not having a lot of dates. also can we make this API expect `[Dates]` from frontend itself and return data as `[Maybe Stats]` back, defaults can be handled at frontend side.

Also as we are planning to make it cached at frontend so if there is only a few days data to be fetched we can just send that date in the API and get the data for only those dates, i.e. when a week is ongoing then we can just fetch the new data everyday...?